x-pack/platform/plugins/shared/streams/server/lib/saved_objects/significant_events/promps_config_service.ts

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type {
+  Logger,
+  SavedObjectsClientContract,
+  SavedObjectsFindResponse,
+  SavedObjectsCreateOptions,
+  SavedObjectsUpdateResponse,
+} from '@kbn/core/server';
+import { systemPromptDefault } from '@kbn/streams-ai/src/significant_events/prompt';
+import { significantEventsPromptsType } from './prompts_config';
+
+const SINGLETON_PROMPTS_ID = 'significant-events-prompts-config-id';
+
+export class PromptsConfigService {
+  private readonly soClient: SavedObjectsClientContract;
+  private readonly logger: Logger;
+
+  constructor({ soClient, logger }: { soClient: SavedObjectsClientContract; logger: Logger }) {
+    this.soClient = soClient;
+    this.logger = logger;
+  }
+
+  /**
+   * Create a new prompt saved object.
+   * attributes is a plain object (e.g. { name, systemPromptTemplate, userPromptTemplate, inputExample })
+   * Note: no forced singleton id/overwrite — allow multiple prompt objects (user-created).
+   */
+  async createPrompt(
+    attributes: {
+      systemPromptTemplate?: string;
+      significantEventsPromptTemplate?: string;
```

Comment:
What is the difference between these two?

I cannot see where we use `significantEventsPromptTemplate` beyond being able to write it, is this for the user prompt?